src/components/BillFilter/__tests__/BillFilter.test.jsx

```diff
@@ -1,134 +1,198 @@
+            categoryOpened: false,
+            dateOpened: false,
+          }}
+          filterToggleChange={mockFilterToggleChange}
+          handleDateSelection={mockHandleDateSelection}
+          handleStatusChange={mockHandleStatusChange}
+        />
+      );
+    });
 
-  beforeEach(() => {
-    mockApplyFiltering = jest.fn();
-    mockCloseFunction = jest.fn();
-    mockDateCheckboxHandler = jest.fn();
-    mockFilterToggleChange = jest.fn();
-    mockHandleDateSelection = jest.fn();
-    mockHandleStatusChange = jest.fn();
+    afterEach(()=> {
+      mockApplyFiltering.mockRestore();
+      mockCloseHandler.mockRestore();
+      mockDateCheckboxHandler.mockRestore();
+      mockFilterToggleChange.mockRestore();
+      mockHandleDateSelection.mockRestore();
+      mockHandleStatusChange.mockRestore();
+    });
 
-    wrapper = shallow(
-      <BillFilter
-        activeTab={'allBills'}
-        applyFiltering={mockApplyFiltering}
-        currentSorting="Newest"
-        billStatusFilter={SAMPLE_BILL_STATUS}
-        closeHandler={mockCloseFunction}
-        dateCheckboxHandler={mockDateCheckboxHandler}
-        dateFilters={SAMPLE_DATE_FILTER}
-        filterToggles={SAMPLE_FILTER_TOGGLES}
-        filterToggleChange={mockFilterToggleChange}
-        handleDateSelection={mockHandleDateSelection}
-        handleStatusChange={mockHandleStatusChange}
-      />
-    );
-    instance = wrapper.instance();
-  });
+    it("should call onClicks of badge clicks", () => {
+      const dateFilters = {
+        startDate: { selected: true, value: "" },
+        endDate: { selected: false, value: "" },
+      };
 
-  afterEach(() => {
+      wrapper.setProps({ dateFilters });
 
-  })
+      wrapper.find("Badge").at(0).simulate("click");
+      expect(mockApplyFiltering).toBeCalledWith("Newest");
+      wrapper.find("Badge").at(1).simulate("click");
+      expect(mockApplyFiltering).toBeCalledWith("Oldest");
+      wrapper.find("Badge").at(3).simulate("click");
+      expect(mockApplyFiltering).toBeCalledWith("A to Z");
+      wrapper.find("Badge").at(4).simulate("click");
+      expect(mockApplyFiltering).toBeCalledWith("Z to A");
+      expect(mockApplyFiltering).toBeCalledTimes(4);
 
-  describe("render", () => {
-    describe("snapshots 📸", () => {
-      it("BillFilter should match snap shot if none of the filters are active", () => {
-        matches(wrapper);
-      });
-      it("BillFilter should match snap shot if date filter is active", () => {
-        wrapper.setProps({
-          filterToggles: {
-            ...SAMPLE_FILTER_TOGGLES,
-            dateOpened: true
-          }
-        })
-        matches(wrapper);
+      wrapper.find("Badge").at(2).simulate("click");
+      expect(mockFilterToggleChange).toBeCalledWith({
+        statusOpened: true,
+        categoryOpened: false,
+        dateOpened: false,
       });
-      it("BillFilter should match snap shot if status filter is active", () => {
```

Comment:
why do you remove these snaps?

we want to test that the props reflect on the render